### ISSUE TITLE
[QUESTION] Questions regarding anomaly scoring documentation

### ISSUE THREAD

**Issue by tristandijkstra** [unit8co/darts#2660]
**Describe the issue linked to the documentation**
I have trouble completely understanding the PyODScorer and KMeansScorer based on the darts documentation page. For clarification I have dissected the problem in a few smaller questions:

1. In the documentation pages for PyODScorer/KMeansScorer  the data is said to be split into moving windows of size W and the following description is given: "For a series of length N, (N - W + 1)/W subsequences will be generated". Depending if the stride is 1 or W either (N - W + 1) or (N // W) may be correct. Which one is it?
  
2. The TimeSeries subsequence passed to the underlying scorer is of shape (W * D). Does this mean a 2D 'array' of shape (W, D), e.g W by D or a 1D vector of length WxD, e.g. W times D?
 
3. Based on the Sklearn documentation, the k-means clusterer takes data in the shape (n_samples, n_features). Which of the following is correct?
   1. The documentation indicates that it is n_features=W, "applying a score per vector of size W". If this is the case, why was this direction chosen instead of ii.?
   2. In the context of inputted subsequences, I would expect that n_samples = W and n_features = D. Thus if D=1, the value at each point in time is clustered. If this is the case, is the score for each item in the vector aggregated to achieve a single score per window vector?

4. PyOD has the same (n_samples, n_features) input format, Is point 3 the same for the PyODScorer?
 
Thank you for your time.

**Additional context**
Relevant darts documentation pages: https://unit8co.github.io/darts/generated_api/darts.ad.scorers.kmeans_scorer.html
And: https://unit8co.github.io/darts/generated_api/darts.ad.scorers.pyod_scorer.html

Sklearn Kmeans docs: https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html

**Comment by dennisbader** [comment]
Hi @tristandijkstra and sorry for the late response.

Have you had a look at our Anomaly Detection example notebook? It features a small section about the KmeansScorer [here](https://unit8co.github.io/darts/examples/22-anomaly-detection-examples.html#Simple-case:-KMeansScorer) with a picture of the window size W.

1. You're right, there seems to be a mistake in the documentation. It should be `N - W + 1`.
2. The model will receive a 2D array of shape `(N - W + 1, W * D)` per series (concatenated along first axis). The first dim is the number of subsequences from point 1.

    `W * D` depends on the `component_wise` parameter:
   1. for `component_wise=False`: A single model is trained on all target components / columns. `W * D` -> (window size W * number of components/columns in the target series D)
   2. for `component_wise=True`: One dedicated model is trained per target component. `W * D` (window size W * 1 column = W) 
   3. That should be explained in my point 2. The number of samples corresponds to the number of subsequences `N - W + 1`.
   4. Yes

Hope this helps :)

Let me add it to the backlog to update the documentation.
